src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -175,21 +175,29 @@ public void visitToken(DetailAST ast) {
 
         if (rcurly != null) {
             final String violation;
-            if (shouldStartLine) {
-                final String targetSourceLine = getLines()[rcurly.getLineNo() - 1];
-                violation = validate(details, option, true, targetSourceLine);
-            }
-            else {
-                violation = validate(details, option, false, "");
-            }
-
+            violation = validate(details, option,
+                    shouldStartLine, getTargetSourceLine(rcurly.getLineNo() - 1));
             if (!violation.isEmpty()) {
                 log(rcurly, violation, "}", rcurly.getColumnNo() + 1);
             }
         }
     }
 
     /**
+     * Fetches line based on passed line no.
+     * @param lineNo for fetching lines.
+     * @return lines associated with line no.
+     */
+    private String getTargetSourceLine(int lineNo) {
```

Comment:
![image](https://cloud.githubusercontent.com/assets/6566854/23584696/12e84c44-018f-11e7-8f10-0e1a626d0264.png)
See this error i am getting.
@romani